src/main/java/io/cdap/plugin/servicenow/source/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -216,4 +224,39 @@ private String getErrorMessage(String responseBody) {
       return e.getMessage();
     }
   }
+
+  /**
+   * Attempt four times with an exponential delay of 120 seconds to fetch the list of records from ServiceNow table when
+   * RetryableException is thrown          .
+   *
+   * @param tableName The ServiceNow table name
+   * @param startDate The start date
+   * @param endDate The end date
+   * @param offset The number of records to skip
+   * @param limit The number of records to be fetched
+   * @return The list of Map; each Map representing a table row
+   */
+  public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName, String startDate, String endDate,
+                                                                  int offset, int limit) {
+    final List<Map<String, Object>> results = new ArrayList<>();
+    Callable<Boolean> fetchRecords = () -> {
+      results.addAll(fetchTableRecords(tableName, startDate, endDate, offset, limit));
+      return true;
+    };
+
+    Retryer<Boolean> retryer = RetryerBuilder.<Boolean>newBuilder()
+      .retryIfExceptionOfType(RetryableException.class)
+      .withWaitStrategy(WaitStrategies.exponentialWait(120000, TimeUnit.MILLISECONDS))
```

Comment:
Make the wait time and stopAfterAttempt values constants

Reply:
Added constants `WAIT_TIME` and `MAX_NUMBER_OF_RETRY_ATTEMPTS`